### PR TITLE
Refactor AE Content Task Finished method and add spec

### PR DESCRIPTION
**Pull request by GregP**
State Machine task finished method refactoring.

This PR is based on the issue below:
ManageIQ/manageiq#12038

@miq-bot add_label refactoring

### CODE REVIEW

**Comment by tinaafitz** [comment]
Hi @GregP Can you address the rubocop issue?

File: spec/content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished_spec.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    end
+
+    let(:root_hash) do
+      {
+        'miq_provision'    => svc_miq_provision_task,
+        'vmdb_object_type' => vmdb_object_type,
+        'miq_server'       => miq_server
+      }
+    end
+    it "task_finished" do
```
Comment (gmcculloug) [inline spec/content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished_spec.rb]: Add a new line between the `let` block and the start of the test.

File: content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished.rb [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
+            def task_finished
+              final_message = "[#{@handle.root['miq_server'].name}] "
+
+              case @handle.root['vmdb_object_type']
+              when 'service_template_provision_task'
+                final_message += "Service [#{task.destination.name}] Provisioned Successfully"
+                @handle.create_notification(:type => :automate_service_provisioned, :subject => task.destination) if task.miq_request_task.nil?
+              when 'miq_provision'
+                final_message += "VM [#{task.get_option(:vm_target_name)}] "
+                final_message += "IP [#{task.vm.ipaddresses.first}] " if task.vm && !task.vm.ipaddresses.blank?
```
Comment (gmcculloug) [inline content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished.rb]: @GregP Rubocop called out this line which @tinaafitz was referencing in her comment.  Along the the rubocop recommendation to use `present?` the condition can be simplified to:
```ruby
if task.vm&.ipaddresses.present?
```

File: content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished.rb [hunk 3]
```diff
@@ -1,22 +1,53 @@
...
+            def task_finished
+              final_message = "[#{@handle.root['miq_server'].name}] "
+
+              case @handle.root['vmdb_object_type']
+              when 'service_template_provision_task'
+                final_message += "Service [#{task.destination.name}] Provisioned Successfully"
+                @handle.create_notification(:type => :automate_service_provisioned, :subject => task.destination) if task.miq_request_task.nil?
+              when 'miq_provision'
+                final_message += "VM [#{task.get_option(:vm_target_name)}] "
+                final_message += "IP [#{task.vm.ipaddresses.first}] " if task.vm.ipaddresses.present?
```
Comment (gmcculloug) [inline content/automate/ManageIQ/System/CommonMethods/StateMachineMethods.class/__methods__/task_finished.rb]: @GregP Thanks for updating this but you missed the `&` mentioned in my original comment https://github.com/ManageIQ/manageiq-content/pull/589#discussion_r341547317 which adds safe navigation to the condition in place of the multi-part AND condition: `if @task.vm && !@task.vm.ipaddresses.blank?`

See http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby

This should really be: `if task.vm&.ipaddresses.present?`

**Pull request merged**